_overviews/core/programming-collections-generically.md

```diff
@@ -0,0 +1,320 @@
+can decide which collection type they want to produce.
+
+To achieve that we have to use `scala.collection.Factory`:
+
+~~~ scala
+trait Factory[-A, +C] {
+
+  /** @return A collection of type `C` containing the same elements
+    *         as the source collection `it`.
+    * @param it Source collection
+    */
+  def fromSpecific(it: IterableOnce[A]): C
+
+  /** Get a Builder for the collection. For non-strict collection
+    * types this will use an intermediate buffer.
+    * Building collections with `fromSpecific` is preferred
+    * because it can be lazy for lazy collections.
+    */
+  def newBuilder: Builder[A, C]
+}
+~~~
+
+The `Factory[A, C]` trait provides two ways of building a collection `C` from
+elements of type `A`:
+ 
+- `fromSpecific`, converts a source collection of `A` to a collection `C`,
+- `newBuilder`, provides a `Builder[A, C]`.
+
+The difference between these two methods is that the former does not necessarily
+evaluate the elements of the source collection. It can produce a non-strict
+collection type (such as `LazyList`) that does not evaluate its elements unless
+it is traversed. On the other hand, the builder-based way of constructing the
+collection necessarily evaluates the elements of the resulting collection.
+In practice, it is recommended to not eagerly evaluate the elements of the collection.
+
+Finally, here is how we can implement a generator of arbitrary collection types:
+
+~~~ scala
+import scala.collection.Factory
+
+implicit def collection[CC[_], A](implicit
+  genA: Gen[A],
+  factory: Factory[A, CC[A]]
+): Gen[CC[A]] =
+  new Gen[CC[A]] {
+    def get: CC[A] = {
+      val lazyElements =
+        LazyList.unfold(()) { _ =>
+          if (Random.nextInt(100) < 10) None
+          else Some((genA.get, ()))
+        }
+      factory.fromSpecific(lazyElements)
+    }
+  }
+~~~
+
+The implementation uses a lazy source collection of a random size (`lazyElements`).
+Then it calls the `fromSpecific` method of the `Factory` to build the collection
+expected by the user.
+
+Here is an example of use of `collection`:
+
+~~~
+scala> collection[List, Int].get
+res0: List[Int] = List(606179450, -1479909815, 2107368132, 332900044, 1833159330, -406467525, 646515139, -575698977, -784473478, -1663770602)
+
+scala> collection[LazyList, Boolean].get
+res1: LazyList[Boolean] = LazyList(_, ?)
+
+scala> collection[Set, Int].get
+res2: Set[Int] = HashSet(-1775377531, -1376640531, -1009522404, 526943297, 1431886606, -1486861391)
+~~~
+
+## Transforming any collection
+
+In this section we will see how we can implement an `intersperse` operation that can be applied to
+any sequence and returns a sequence with a new element inserted between each element of the
+source sequence.
+
+Transforming collections consists in both consuming and producing collections. This is achieved by
+combining the techniques described in the previous sections. We can already start hacking with
+something like the following:
```

Comment:
I'd suggest swapping this paragraph with the preceding paragraph, and also perhaps expanding it to remind the reader more where we are in the overall progression. (Having dug through the details of the `Gen` example, I've lost context and need help reorienting myself.)